### ISSUE TITLE
Loading via load() method doesn't work 

### ISSUE THREAD

**Issue by RaiserWeb** [staudenmeir/eloquent-json-relations#41]
Hi, great package. I get zero results when I use `$model->load('relation')` , which seems like a bug? EG:

```
// the relation on Booking model:
    public function notifications(){
        return $this->hasMany('App\Models\Notification', 'data->booking_id');
    }

dd($booking->notifications); // this provides results

// if I first load the relation, I get no results
$booking->load('notifications');
dd($booking->notifications); // this provides no results
```

Using v1.5, Laravel 8

**Comment by staudenmeir** [comment]
Can you log the executed queries for `$booking->notifications` and `$booking->load('notifications')`?

Does `Booking::with('notifications')->get()` work correctly?

**Comment by archilex** [comment]
I'm having the same issues with eager/lazy loading. Here are the queries:

Returns an empty array:
```
dd(Email::with('events')->first()->events);

select * from `sendgrid_webhook_events` where json_unquote(json_extract(`sendgrid_webhook_events`.`payload`, '$."uniqid"')) in ('7lTuR7q3KAuf8WLVkXH3nITtaU5KRy')
```

VS

Returns the results:
```
Email::first()->events();

select * from `sendgrid_webhook_events` where json_unquote(json_extract(`sendgrid_webhook_events`.`payload`, '$."uniqid"')) = '7lTuR7q3KAuf8WLVkXH3nITtaU5KRy' and (json_extract(`sendgrid_webhook_events`.`payload`, '$."uniqid"') is not null AND json_type(json_extract(`sendgrid_webhook_events`.`payload`, '$."uniqid"')) != 'NULL')
```

It's worth noting that BOTH queries return results in TablePlus

Using 1.4.3, Laravel 7.29

**Comment by staudenmeir** [comment]
Is `dd(Email::with('events')->get()->first()->events);` also empty?

Are both queries getting the same `email`?

**Comment by archilex** [comment]
Yes, `dd(Email::with('events')->get()->first()->events);` is also empty.

Yes, all the queries are getting the same email. To be sure they were getting the same email I tested like this as well:

**Returns empty array:**
Filtering the collection: `dd(Email::with('events')->get()->where('id', 100)->first()->events);`
Filtering the query: `dd(Email::with('events')->firstWhere('id', 100)->events);`

(Both those queries return the appropriate model without `->events`)

**Returns results:**
`dd(Email::find(100)->events);`
`dd(Email::firstWhere('id', 100)->events);`

So basically I only get results when not eager loading the relationship.

**Comment by staudenmeir** [comment]
Can you share the `Email` and `Event` models?

**Comment by archilex** [comment]
Sorry. My issue was user error. I did not add the trait to both models as specified in the documentation. I had read your line reminding people to use it in both models, however since I didn't have a need for accessing the relationship from related `Event` model, I assumed adding the trait was not required. After adding the trait in the related `Event` model eager loading worked as expected.

If anyone else runs into this issue, the relationship is not required BUT the trait **is required**. (Just like the docs say!) 😂

Thanks for a great package!
